src/nvim/event/socket.c

```diff
@@ -103,10 +101,11 @@ int socket_watcher_start(SocketWatcher *watcher, int backlog, socket_cb cb)
         // contain 0 in this case, unless uv_tcp_getsockname() is used first.
         uv_tcp_getsockname(&watcher->uv.tcp.handle, (struct sockaddr *)&sas,
                            &(int){ sizeof(sas) });
-        uint16_t port = (uint16_t)(
-                                   (sas.ss_family == AF_INET)
-            ? (STRUCT_CAST(struct sockaddr_in, &sas))->sin_port
-            : (STRUCT_CAST(struct sockaddr_in6, &sas))->sin6_port);
+        uint16_t port =
+          (uint16_t)((sas.ss_family ==
+                      AF_INET) ? (STRUCT_CAST(struct sockaddr_in,
```

Comment:
Something went wrong here. Ideally the two `STRUCT_CAST` should (at least almost) align